.github/workflows/e2e-testing.yaml

```diff
@@ -61,9 +61,42 @@ jobs:
           MONGODB_ATLAS_ORG_ID: ${{ secrets.CLOUD_DEV_ORG_ID }}
           MONGODB_ATLAS_BASE_URL: https://cloud-dev.mongodb.com/
           MONGODB_ATLAS_SECRET_PROFILE: cfn-cloud-dev-github-action
+        # run idividual test allowing for parallel execution. Due to usage of t.Setenv() in test code, t.Parallel() is not possible
         run: |
           cd cfn-resources/test/e2e/cluster
-          go test -timeout 90m -v cluster_test.go
+          go test -timeout 90m -v -run '^TestClusterCFN$' .
+
+  cluster-pause:
+    needs: change-detection
+    if: ${{ needs.change-detection.outputs.cluster == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8
+      - uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+      - run: pip install cloudformation-cli-go-plugin
+      - uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00
+        with:
+          go-version-file: 'cfn-resources/go.mod'
+      - uses: aws-actions/configure-aws-credentials@00943011d9042930efac3dcd3a170e4273319bc8
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID_TEST_ENV }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY_TEST_ENV }}
+          aws-region: eu-west-1
+      - name: Run E2E test
+        shell: bash
+        env:
+          MONGODB_ATLAS_PUBLIC_KEY: ${{ secrets.CLOUD_DEV_PUBLIC_KEY }}
+          MONGODB_ATLAS_PRIVATE_KEY: ${{ secrets.CLOUD_DEV_PRIVATE_KEY }}
+          MONGODB_ATLAS_ORG_ID: ${{ secrets.CLOUD_DEV_ORG_ID }}
+          MONGODB_ATLAS_BASE_URL: https://cloud-dev.mongodb.com/
+          MONGODB_ATLAS_SECRET_PROFILE: cfn-cloud-dev-github-action
+        # run idividual test allowing for parallel execution. Due to usage of t.Setenv() in test code, t.Parallel() is not possible
+        run: |
+          cd cfn-resources/test/e2e/cluster
+          go test -timeout 90m -v -run '^TestClusterPauseCFN$' .    
```

Comment:
While not ideal this was a solution that did not require a significant refactor to how our CFN e2e are currently executed and allow the new test to run in parallel. Currently e2e tests rely on executing shell scripts and setting env variables during execution.